packages/git/src/browser/git-decorator.ts

```diff
@@ -73,7 +74,8 @@ export class GitDecorator implements TreeDecorator {
             return result;
         }
         const markers = this.appendContainerChanges(tree, status.changes);
-        for (const { id } of new DepthFirstTreeIterator(tree.root)) {
+        for (const treeNode of new DepthFirstTreeIterator(tree.root)) {
+            const id = FileStatNode.getUri(treeNode) || treeNode.id;
```

Comment:
it looks suspicious, markers uses URI as a key, results should have node ID as key, not URI, here `id` seems to try to server for both of them